myia/api.py

```diff
@@ -100,3 +107,97 @@ def myfun2(cond, x, y):
     return MyiaFunction(fn, specialize_values, backend=backend,
                         backend_options=backend_options,
                         return_backend=return_backend)
+
+
+######################################################################
+# Converts args to initialize model on target accelerator hardware   #
+# Note: not to be conflated with weight initialization distributions #
+######################################################################
+
+@overload(bootstrap=True)
+def _convert_arg_init(self, arg, orig_t: AbstractTuple, backend):
+    if not isinstance(arg, tuple):
+        raise TypeError('Expected tuple')
+    oe = orig_t.elements
+    if len(arg) != len(oe):
+        raise TypeError(f'Expected {len(oe)} elements')
+    return tuple(self(x, o, backend) for x, o in zip(arg, oe))
+
+
+@overload  # noqa: F811
+def _convert_arg_init(self, arg, orig_t: AbstractList, backend):
+    if not isinstance(arg, list):
+        raise TypeError('Expected list')
+    ot = orig_t.element
+    return [self(x, ot, backend) for x in arg]
+
+
+@overload  # noqa: F811
+def _convert_arg_init(self, arg, orig_t: AbstractClass, backend):
+    if not isinstance(arg, orig_t.tag):
+        raise TypeError(f'Expected {orig_t.tag.__qualname__}')
+    arg = tuple(getattr(arg, attr) for attr in orig_t.attributes)
+    oe = list(orig_t.attributes.values())
+    return orig_t.tag(*(self(x, o, backend) for x, o in zip(arg, oe)))
+
+
+@overload  # noqa: F811
+def _convert_arg_init(self, arg, orig_t: AbstractArray, backend):
+    et = orig_t.element
+    assert isinstance(et, AbstractScalar)
+    et = et.values[TYPE]
+    assert issubclass(et, dtype.Number)
+    if isinstance(arg, np.ndarray):
+        arg = ArrayWrapper(backend.from_numpy(arg), arg.dtype, arg.shape)
+    backend.check_array(arg.array, et)
+    return arg
+
+
+"""
+TODO: AbstractUnion overload of _convert_arg_init().
+
+This overload will look similar to AbstractUnion overload
+from convert_arg() in myia/pipeline/steps.py
+
+AbstractUnion overload from convert_arg() at time of this commit:
+https://github.com/mila-iqia/myia/blob/
+c350b341f52d2d0e3dc1e5ab1d890103a45b60c9/
+myia/pipeline/steps.py#L528-L537
+
+Current AbstractUnion overload from convert_arg():
+(Note: might have been moved to different location of codebase
+since this comment was written/committed):
+https://github.com/mila-iqia/myia/blob/master/myia/pipeline/steps.py#L528-L537
+"""
```

Comment:
I don't think this one will be needed since you can't pass in a value that is of union type.  I don't see much harm in leaving the comment though.